File: cropme/src/main/java/com/takusemba/cropme/CropImageView.java
package com.takusemba.cropme;

import android.content.Context;
import android.graphics.RectF;
import android.support.v7.widget.AppCompatImageView;
import android.util.AttributeSet;

/**
 * ImageView to hold an Image to be animated.
 **/
class CropImageView extends AppCompatImageView {

    private RectF frame;

    public CropImageView(Context context) {
        this(context, null);
    }

    public CropImageView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public CropImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    /**
     * set image size along with {@link CropImageView#frame}
     **/
    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        float height = getMeasuredHeight();
        float width = getMeasuredWidth();
        float widthScale = frame != null ? frame.width() / width : 1;
        float heightScale = frame != null ? frame.height() / height : 1;
        float scale = Math.max(widthScale, heightScale);
        setMeasuredDimension((int) (width * scale), (int) (height * scale));
    }

    void setFrame(RectF frame) {
        this.frame = frame;
    }
}

File: cropme/src/main/java/com/takusemba/cropme/CropImageView.kt
package com.takusemba.cropme

import android.content.Context
import android.graphics.RectF
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

/**
 * ImageView to show a image for cropping.
 */
class CropImageView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : AppCompatImageView(context, attrs, defStyleAttr) {

  private var frame: RectF? = null

  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
    super.onMeasure(widthMeasureSpec, heightMeasureSpec)
    val height = measuredHeight.toFloat()
    val width = measuredWidth.toFloat()
    val frameRect = this.frame
    val widthScale = if (frameRect != null) frameRect.width() / width else 1f
    val heightScale = if (frameRect != null) frameRect.height() / height else 1f
    val scale = maxOf(widthScale, heightScale)
    setMeasuredDimension((width * scale).toInt(), (height * scale).toInt())
  }

  fun setFrame(frame: RectF) {
    this.frame = frame
  }
}
